pycontrails/datalib/goes.py

```diff
@@ -0,0 +1,730 @@
+        The Geostationary projection built from the GOES metadata.
+    src_extent : tuple[float, float, float, float]
+        Extent of GOES data in the Geostationary projection
+    """
+    proj_info = da.attrs["goes_imager_projection"]
+    h = proj_info["perspective_point_height"]
+    lon0 = proj_info["longitude_of_projection_origin"]
+    src_crs = ccrs.Geostationary(central_longitude=lon0, satellite_height=h, sweep_axis="x")
+
+    if color_scheme == "true":
+        rgb = to_true_color(da, gamma)
+    elif color_scheme == "ash":
+        rgb = to_ash(da, ash_convention)
+    else:
+        raise ValueError(f"Color scheme must be 'true' or 'ash', not '{color_scheme}'")
+
+    np.nan_to_num(rgb, copy=False)
+
+    x = da["x"].values
+    y = da["y"].values
+
+    # Multiply extremes by the satellite height
+    src_extent = h * x.min(), h * x.max(), h * y.min(), h * y.max()
+
+    return rgb, src_crs, src_extent
+
+
+def to_true_color(da: xr.DataArray, gamma: float = 2.2) -> npt.NDArray[np.float32]:
+    """Compute 3d RGB array for the true color scheme.
+
+    Parameters
+    ----------
+    da : xr.DataArray
+        DataArray of GOES data with channels C01, C02, C03.
+    gamma : float = 2.2
+        Gamma correction for the RGB channels.
+
+    Returns
+    -------
+    npt.NDArray[np.float32]
+        3d RGB array with true color scheme.
+
+    References
+    ----------
+    - `Unidata's true color recipe <https://unidata.github.io/python-gallery/examples/mapping_GOES16_TrueColor.html>`_
+    """
+    red = da.sel(band_id=2).values
+    green = da.sel(band_id=3).values
+    blue = da.sel(band_id=1).values
+
+    red = _clip_and_scale(red, 0.0, 1.0)
+    green = _clip_and_scale(green, 0.0, 1.0)
+    blue = _clip_and_scale(blue, 0.0, 1.0)
+
+    red = red ** (1 / gamma)
+    green = green ** (1 / gamma)
+    blue = blue ** (1 / gamma)
+
+    # Calculate "true" green channel
+    green = 0.45 * red + 0.1 * green + 0.45 * blue
+    green = _clip_and_scale(green, 0.0, 1.0)
+
+    return np.dstack([red, green, blue])
+
+
+def to_ash(da: xr.DataArray, convention: str = "MIT") -> npt.NDArray[np.float32]:
+    """Compute 3d RGB array for the ASH color scheme.
+
+    Parameters
+    ----------
+    da : xr.DataArray
+        DataArray of GOES data with appropriate channels.
+    convention : str = {"MIT", "standard"}
+        Convention for color space.
+
+        - MIT convention requires channels C11, C14, C15
+        - Standard convention requires channels C11, C13, C14, C15
+
+    Returns
+    -------
+    npt.NDArray[np.float32]
+        3d RGB array with ASH color scheme according to convention.
+
+    References
+    ----------
+    - `Ash RGB quick guide (the color space and color interpretations) <https://rammb.cira.colostate.edu/training/visit/quick_guides/GOES_Ash_RGB.pdf>`_
+    - `MIT thesis detail use for contrail detection <https://dspace.mit.edu/bitstream/handle/1721.1/124179/1144176218-MIT.pdf?22>`_
```

Comment:
We should cite this in the Zotero, I'll push to this branch

Reply:
kulikSatellitebasedDetectionContrails2019 (0836a5a4649f59ac29e340d25df0d9645536dea9)

Reply:
481d223b7a633578209b1f10fd5f3d5d9c8fc8b1